src/test/java/mate/academy/spring/boot/serviceTests/BookServiceTests.java

```diff
@@ -0,0 +1,184 @@
+        when(bookMapper.toDto(any())).thenReturn(expected.get(0));
+        List<BookDto> actual = bookService.findAll(pageable);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+            Test checks whether the method correctly
+            calls the delete method by ID to the repository""")
+    void testDeleteById_WithCorrectId_ShouldCallRepository() {
+        Long id = 1L;
+        bookService.deleteById(id);
+        verify(bookRepository).deleteById(id);
+    }
+
+    @Test
+    @DisplayName("""
+            This test tests a service method that should
+            update the book data and return the updated BookDto object.""")
+    void testUpdate_WithCorrectId_ShouldReturnUpdatedBookDto() {
+        Book book = new Book();
+        UpdateBookRequestDto requestDto = new UpdateBookRequestDto();
+        BookDto expected = new BookDto();
+        when(bookRepository.findById(1L)).thenReturn(Optional.of(book));
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toDto(book)).thenReturn(expected);
+        BookDto actual = bookService.update(1L, requestDto);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+            The test checks should determine
+            if an empty list is returned when we search
+            for books by a nonexistent or empty category""")
+    void testFindAllByCategoryId_WithNonExistentCategory_ShouldReturnEmptyList() {
+        Long categoryId = 1L;
+        when(bookRepository.findAllByCategoryId(categoryId)).thenReturn(List.of());
+        List<BookDtoWithoutCategoryIds> actual = bookService.findAllByCategoryId(categoryId);
+        assertTrue(actual.isEmpty());
+    }
+
+    @Test
+    @DisplayName("""
+            Test checks whether the searchBooks method
+            correctly searches and returns a list of books based on the specified search parameters""")
+    void testSearchBooks_WithCorrectParameters_ShouldReturnBooks() {
+        BookSearchParameters parameters = new BookSearchParameters(
+                new String[]{"Author"},
+                new String[]{"Category"},
+                new String[]{"Title"}
+        );
+        Pageable pageable = PageRequest.of(0, 10);
+        Book book = new Book();
+        BookDto bookDto = new BookDto();
+        Specification<Book> spec = mock(Specification.class);
+        Page<Book> bookPage = new PageImpl<>(List.of(book), pageable, 1);
+        when(bookSpecificationBuilder.build(parameters)).thenReturn(spec);
+        when(bookRepository.findAll(spec, pageable)).thenReturn(bookPage);
+        when(bookMapper.toDto(any(Book.class))).thenReturn(bookDto);
+        List<BookDto> actual = bookService.searchBooks(parameters, pageable);
+        List<BookDto> expected = List.of(bookDto);
+        assertEquals(expected, actual);
+    }
+
+    private CreateBookRequestDto getCreateBookRequestDto(String title, String author,
```

Comment:
You can simplify this method and use the same values for your `requestDto`, not passing them in your test methods
The same with other private methods with test data (where it's possible)